src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -21,10 +22,28 @@ async function assume_role(req) {
         }
         throw new StsError(StsError.InternalFailure);
     }
+
+    let seconds_until_expiration;
+    // Check if duration_seconds was provided via the CLI, otherwise resort to the default
+    if (req.body.duration_seconds) {
+        // duration_seconds is a string, convert it to a number
+        const duration_seconds = Number(req.body.duration_seconds);
+        // If it was provided, check if it's within the allowed range dictated by Amazon
+        if (duration_seconds < 900 || duration_seconds > 43200) {
+            dbg.error('Value', duration_seconds, 'for duration_seconds is not within the allowed range of 900-43200 seconds');
+            throw new StsError(StsError.ValidationError);
```

Comment:
I think you can improve the error message - you can leave it as a gap and handle it in the future.

The general message is 'The input fails to satisfy the constraints specified by an AWS service.'
But in real cases, you see it is a specific message to the case, in your example:
"An error occurred (ValidationError) when calling the AssumeRole operation: 1 validation error detected: Value '43201' at 'durationSeconds' failed to satisfy constraint: Member must have a value less than or equal to 43200".
I remembered that @guymguym told me that we are compatible with AWS, but being compatible with the error message is not something we have to... In my opinion is it more user-friendly.

I did something like that for IAM in case you want. In the future I can move the function to a higher module in the stuck for reuse.
https://github.com/noobaa/noobaa-core/blob/8b634ddf35aaf3789539bd7f215c0bc09640d218/src/endpoint/iam/iam_utils.js#L104-L119